src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -244,16 +245,23 @@ public void setConfig(File file) {
     // -@cs[AbbreviationAsWordInName] Should be removed at 7.0 version,
     // we keep for some time to avoid braking compatibility.
     @Deprecated
-    public void setConfigURL(URL url) {
+    public void setConfigURL(String url) {
         setConfigUrl(url);
     }
 
     /**
      * Sets URL to the configuration.
      * @param url the URL of the configuration to use
      */
-    public void setConfigUrl(URL url) {
-        setConfigLocation(url.toExternalForm());
+    public void setConfigUrl(String url) {
+        try {
+            setConfigLocation(new URL(url).toExternalForm());
```

Comment:
we are decided to go further with breaking compatibility changes, please read https://github.com/checkstyle/checkstyle/issues/4449#issuecomment-310078283